src/station_cmd.cpp

```diff
@@ -3802,6 +3804,13 @@ void IncreaseStats(Station *st, const Vehicle *front, StationID next_station_id)
 			 * among the wagons in that case.
 			 * As usage is not such an important figure anyway we just
 			 * ignore the additional cargo then.*/
+			Station *next_station = Station::Get(next_station_id);
+			if (current_order_is_implicit && !HasBit(GetAcceptanceMask(next_station), v->cargo_type)) {
+				/* Prevent node links to be created for implicit stops
+				 * whose stations do not accept the provided cargo. */
+				break;
```

Comment:
`break` or `continue` ? (what if there is a wagon with cargo that is accepted by the station?). I don't know much about cargodist, so this really is a question; I can be completely wrong!